spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/CreateChangelogViewProcedure.java

```diff
@@ -228,22 +247,41 @@ private Dataset<Row> applyChangelogIterator(Dataset<Row> df, Column[] repartitio
             RowEncoder.apply(schema));
   }
 
-  private Dataset<Row> applyCarryoverRemoveIterator(Dataset<Row> df, Column[] repartitionSpec) {
-    Column[] sortSpec = sortSpec(df, repartitionSpec);
+  private Dataset<Row> applyCarryoverRemoveIterator(
+      Dataset<Row> df, Column[] repartitionSpec, boolean netChanges) {
+    Column[] sortSpec = sortSpec(df, repartitionSpec, netChanges);
     StructType schema = df.schema();
 
     return df.repartition(repartitionSpec)
         .sortWithinPartitions(sortSpec)
         .mapPartitions(
             (MapPartitionsFunction<Row, Row>)
-                rowIterator -> ChangelogIterator.removeCarryovers(rowIterator, schema),
+                rowIterator ->
+                    netChanges
+                        ? ChangelogIterator.removeNetCarryovers(rowIterator, schema)
+                        : ChangelogIterator.removeCarryovers(rowIterator, schema),
             RowEncoder.apply(schema));
   }
 
-  private static Column[] sortSpec(Dataset<Row> df, Column[] repartitionSpec) {
-    Column[] sortSpec = new Column[repartitionSpec.length + 1];
-    System.arraycopy(repartitionSpec, 0, sortSpec, 0, repartitionSpec.length);
+  private static Column[] sortSpec(Dataset<Row> df, Column[] repartitionSpec, boolean netChanges) {
+    Column[] sortSpec;
+
+    if (netChanges) {
```

Comment:
It looks a bit hard to read, what do you think like:


int[] generatedColumns = netChanges ? new Column[] {changeOrdinalCol, changeTypeCol} : new Column[] { changeTypeCol };

Column[] sortSpec = new Column[repartitionSpec.length + generatedColumns.length];
System.arrayCopy(repartitionSpec, 0, sortSpec, 0, ...);
System.arrayCopy(generatedColumns, 0, sortSpec, repartitionSpec.length, repartitionSpec.length + generatedColumns.length);

Maybe can shorten the variable names to make it shorter.

Reply:
Made the change, thanks for the suggestion